From a59f44c1bc19e9092fc31337e60f03eb7e9ac5c9 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 19 Apr 2021 10:01:41 +0300
Subject: [PATCH] [SW-37085] gaudi: security: rr max addr update

Normally for LBW RRs min<hit<max is the
configuration norm followed for min and max
address ranges. The hit address is not inclusive
of min and max, hence (min-1) and (max+1)
are configured respectively as range addresses.

But for RR 'LBW_RR_DBG_ETR' the max address is
0x7F_FFFF_FFFF, for which (max+1) would not be
a valid address location. As 0x7F_FFFF_FFFF is
the last addressable location in gaudi1 the max
address will be configured as max and not (max+1).

This is special case exception for LBW RR configuration.
Due to this the last byte will not be included in the
hit range.

Change-Id: I14790b7527cdcfbe434a34ffbcaa9e687e43cf94
Signed-off-by: Karthikeyan K <karthikk@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index ff38eb6..7ea9874 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1370,7 +1370,16 @@
 	 */
 	/* Debug address space : 0x7FFE000000 - 0x7FFFFFFFFF */
 	min = mmMME_S_ROM_TABLE_BASE - 1;
-	max = mmTPC7_EML_CS_BASE + TPC7_EML_CS_MAX_OFFSET;
+	/*
+	 * Spl case: As per LBW RR config 'min<hit<max' which means
+	 * address ranges of min-1 and max+1 should be configured, as
+	 * hit is not inclusive. But for DBG address max address is 0x7FFFFFFFFF
+	 * and max+1(0x80_0000_0000) is not a valid location.
+	 * Hence max+1 logic should not be applied for this case alone
+	 * and the max address is configured as 0x7FFFFFFFFF and not
+	 * 0x80_0000_0000.
+	 */
+	max = mmTPC7_EML_CS_BASE + TPC7_EML_CS_MAX_OFFSET - 1;
 	preboot_secure_lbw_rr_if(LBW_RR_DBG_ETR, min, max);
 
 	/* SYNC_MNGR: PCIe only! PCIe RR_28 up to PCIe RR_31 */
